crypto/awskms/awskms.go

```diff
@@ -0,0 +1,171 @@
+/*
+ * Flow Go SDK
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package awskms provides a AWS Key Management Service (KMS)
+// implementation of the crypto.Signer interface.
+//
+// The documentation for AWS KMS can be found here: https://docs.aws.amazon.com/kms/index.html
+package awskms
+
+import (
+	"context"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	kms "github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/onflow/flow-go-sdk/crypto"
+)
+
+const (
+	resouceArnFormat        = "arn:aws:kms:%s:%s:key/%s"
+	resourceIDArgumentCount = 5
+)
+
+// Key is a reference to a AWS KMS asymmetric signing key.
+type Key struct {
+	Region  string `json:"region"`
+	Account string `json:"account"`
+	KeyID   string `json:"keyId"`
+}
+
+// ARN returns the KMS arn for this KMS key.
+// For cross account key access, you need to pass the arn instead of just the keyID.
+func (k Key) ARN() string {
+	return fmt.Sprintf(
+		resouceArnFormat,
+		k.Region,
+		k.Account,
+		k.KeyID,
+	)
+}
+
+// Example ARN format: "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"
+func KeyFromResourceARN(resourceARN string) (Key, error) {
+	key := Key{}
+	spiltedARN := strings.Split(resourceARN, ":")
+	if len(spiltedARN) != 6 {
+		return key, fmt.Errorf("awskms: wrong format for the resourceARN: %s", resourceARN)
+	}
+
+	key.Region, key.Account = spiltedARN[3], spiltedARN[4]
+	key.KeyID = strings.Split(spiltedARN[5], "/")[1]
+
+	return key, nil
+}
+
```

Comment:
I feel like this is a bit redundant no?

Because the KeyFromResourceARN is already being passed the KMS ARN in it's entirety. And I don't see the other files using the region or account ID as separate attributes.

Reply:
yeah I think about it, but still keep it there for the reasons:
1. clients can use multi-region key, so clients can provide different region with the same keyId. 
2. similar code structure as existing cloudkms